packages/payment-processor/src/payment/index.ts

```diff
@@ -25,28 +32,47 @@ export class UnsupportedNetworkError extends Error {
   }
 }
 
+/**
+ * Details required for a token swap:
+ *
+ *  - maxInputAmount: maximum number of ERC20 allowed for the swap before payment, considering both amount and fees
+ *  - path: array of token addresses to be used for the "swap path".
+ *    ['0xPaymentCurrency', '0xIntermediate1', ..., '0xRequestCurrency']
+ *  - deadline: time in milliseconds since UNIX epoch, after which the swap should not be executed.
+ */
+export interface ISwapSettings {
+  deadline: number;
+  maxInputAmount: BigNumber;
+  path: string[];
+}
+
 /**
  * Processes a transaction to pay a Request.
  * Supported networks: ERC20_PROXY_CONTRACT, ETH_INPUT_DATA, ERC20_FEE_PROXY_CONTRACT
  *
- * @throws UnsupportedNetworkError if network isn't supported
+ * @throws UnsupportedNetworkError if network isn't supported for swap or payment.
  * @param request the request to pay.
  * @param signerOrProvider the Web3 provider, or signer. Defaults to window.ethereum.
  * @param amount optionally, the amount to pay. Defaults to remaining amount of the request.
+ * @param swapSettings optionally, the information of how to swap from another payment token.
  * @param overrides optionally, override default transaction values, like gas.
  */
 export async function payRequest(
   request: ClientTypes.IRequestData,
   signerOrProvider: Web3Provider | Signer = getProvider(),
   amount?: BigNumberish,
+  swapSettings?: ISwapSettings,
   overrides?: ITransactionOverrides,
 ): Promise<ContractTransaction> {
   const signer = getSigner(signerOrProvider);
   const paymentNetwork = getPaymentNetwork(request);
+  if (swapSettings && !canSwapToPay(request)) {
+    throw new UnsupportedNetworkError(paymentNetwork);
```

Comment:
I would add a more specific error messsage... Especially if we keep the breaking change, since this error might happen to people upgrading the library